pkg/sdk/identifier_helpers_test.go

```diff
@@ -82,3 +83,42 @@ func TestDatabaseObjectIdentifier(t *testing.T) {
 		assert.Equal(t, `"aaa"."bbb"`, identifier.FullyQualifiedName())
 	})
 }
+
+func TestNewSchemaObjectIdentifierWithArgumentsFromFullyQualifiedName(t *testing.T) {
+	testCases := []struct {
+		RawInput string
+		Input    SchemaObjectIdentifierWithArguments
+		Error    string
+	}{
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, DataTypeNumber, DataTypeTimestampTZ)},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, "VECTOR(INT, 20)")},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, "VECTOR(INT, 20)", DataTypeFloat)},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, "VECTOR(INT, 20)", "VECTOR(INT, 10)")},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeTime, "VECTOR(INT, 20)", "VECTOR(FLOAT, 10)", DataTypeFloat)},
+		// TODO(): Won't work, because of the assumption that identifiers are not containing '(' and ')' parentheses
+		{Input: NewSchemaObjectIdentifierWithArguments(`ab()c`, `def()`, `()ghi`, DataTypeTime, "VECTOR(INT, 20)", "VECTOR(FLOAT, 10)", DataTypeFloat), Error: `unable to read identifier: "ab`},
+		{Input: NewSchemaObjectIdentifierWithArguments(`ab(,)c`, `,def()`, `()ghi,`, DataTypeTime, "VECTOR(INT, 20)", "VECTOR(FLOAT, 10)", DataTypeFloat), Error: `unable to read identifier: "ab`},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`)},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`), RawInput: `abc.def.ghi()`},
+		{Input: NewSchemaObjectIdentifierWithArguments(`abc`, `def`, `ghi`, DataTypeFloat, "VECTOR(INT, 20)"), RawInput: `abc.def.ghi(FLOAT, VECTOR(INT, 20))`},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(fmt.Sprintf("processing %s", testCase.Input.FullyQualifiedName()), func(t *testing.T) {
+			var id SchemaObjectIdentifierWithArguments
+			var err error
+			if testCase.RawInput != "" {
```

Comment:
This comment I have already left several times: it's better not to have too much conditional logic in the parametrized tests, it's error-prone and harder to read. It's better to split into several parametrized ones, here e.g. to input, raw input, and invalid.

Reply:
Moved tests to `identifier_parsers_test.go` (as we're using ParseXxxIdentifier now) and split this test into two